controllers/activemqartemis_reconciler.go

```diff
@@ -1040,6 +1038,25 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) applyTemplate(index int, templa
 	return nil
 }
 
+func formatTemplatedObject(customResource *brokerv1beta1.ActiveMQArtemis, object interface{}, ordinal string, itemName string, resType string) interface{} {
+	if objectMap, isObjectMap := object.(map[string]interface{}); isObjectMap {
+		targetMap := make(map[string]interface{})
+		for objectMapKey, objectMapValue := range objectMap {
+			targetMap[objectMapKey] = formatTemplatedObject(customResource, objectMapValue, ordinal, itemName, resType)
+		}
+		return targetMap
+	} else if objectArray, isObjectArray := object.([]interface{}); isObjectArray {
+		targetArray := make([]interface{}, len(objectArray))
+		for objectArrayIndex, objectArrayValue := range objectArray {
+			targetArray[objectArrayIndex] = formatTemplatedObject(customResource, objectArrayValue, ordinal, itemName, resType)
+		}
+		return targetArray
+	} else if objectString, isObjectString := object.(string); isObjectString {
+		return formatTemplatedString(customResource, objectString, ordinal, itemName, resType)
+	}
+	return object
+}
+
```

Comment:
The recursive formatTemplatedObject creates new objects at every level. For large patches with many resources, this could be memory-intensive. However this could be a very minor problem if we only expect small patches.

Reply:
This should not be an issue because we expect only small patches and the formatted patch goes out of scope and becomes unreachable when applyTemplate returns. At this point, the related memory becomes eligible for collection by Go's garbage collector, which will automatically reclaim it during its next collection cycle.